source/geometries/Next100Shielding.cc

```diff
@@ -343,123 +550,224 @@ namespace nexus {
       	  }
-      	}
-      	// VertexVolume = geom_navigator_->LocateGlobalPointAndSetup(vertex, 0, false);
-      	// } while (VertexVolume->GetName() != "STEEL_BEAM_ROOF");
+      	  else{
+      	    vertex = ped_front_gen_->GenerateVertex("INSIDE");
+      	    vertex.setZ(vertex.z() - (2.*support_front_dist_ + support_beam_dist_));
+      	  }
+        }
+        else if (rand < (perc_ped_bottom_vol_ + perc_ped_top_vol_ + perc_ped_front_vol_ + perc_ped_lateral_vol_)){ // LATERAL BEAM
+          if (G4UniformRand() < 0.5) {
+            vertex = ped_lateral_gen_->GenerateVertex("INSIDE");
+          }
+          else{
+            vertex = ped_lateral_gen_->GenerateVertex("INSIDE");
+            vertex.setX(vertex.x() - (pedestal_x_ + pedestal_lateral_beam_thickness));
+          }
+         }
+        else { // ROOF
+          if (G4UniformRand() < 0.5) {
+            if (G4UniformRand() < 0.5){
+              vertex = ped_roof_lat_gen_->GenerateVertex("INSIDE");
+            }
+            else{
+              vertex = ped_roof_lat_gen_->GenerateVertex("INSIDE");
+              vertex.setX(vertex.x() - pedestal_top_x_ - pedestal_roof_thickness);
+            }
+          }
+          else{
+            if (G4UniformRand() < 0.5){
+              vertex = ped_roof_front_gen_->GenerateVertex("INSIDE");
+            }
+            else{
+              vertex = ped_roof_front_gen_->GenerateVertex("INSIDE");
+              vertex.setZ(vertex.z() - pedestal_lateral_length + pedestal_roof_thickness);
+            }
+          }
+         }
       }
 
-      else if (rand < perc_top_struct_vol_) { //TOP BEAM STRUCTURE
-	G4double random = G4UniformRand();
-	if (random <  perc_struc_x_vol_){
-	  G4double rand_beam = int (4* G4UniformRand());
-	  if (rand_beam == 0) {
-	    vertex = struct_x_gen_->GenerateVertex("INSIDE");
-	  }
-	  else if (rand_beam == 1) {
-	    vertex = struct_x_gen_->GenerateVertex("INSIDE");
-	    vertex.setZ(vertex.z()-roof_z_separation_);
-	  }
-	  else if (rand_beam == 2) {
-	    vertex = struct_x_gen_->GenerateVertex("INSIDE");
-	    vertex.setZ(vertex.z()-(roof_z_separation_+lateral_z_separation_));
-	  }
-	  else if (rand_beam == 3) {
-	    vertex = struct_x_gen_->GenerateVertex("INSIDE");
-	    vertex.setZ(vertex.z()-(2*roof_z_separation_+lateral_z_separation_));
-	  }
-	}
-	else {
-	  if (G4UniformRand() < 0.5) {
-	    vertex = struct_z_gen_->GenerateVertex("INSIDE");
-	  }
-	  else {
-	    vertex = struct_z_gen_->GenerateVertex("INSIDE");
-	    vertex.setX(vertex.x()+front_x_separation_);
-	  }
-	}
+    else if(region=="BUBBLE_SEAL"){
+      G4double rand = G4UniformRand();
+      if (rand<perc_bubble_front_vol_){ // front
+        if (G4UniformRand() < 0.5){
+          vertex = bubble_seal_front_gen_->GenerateVertex("INSIDE");
+          vertex.setZ(vertex.z() + (support_beam_dist_/2. + support_front_dist_ + pedestal_front_beam_thickness/2. + bubble_seal_thickness/2.));
```

Comment:
To simplify the code further and make it more readable, you can put the value to add or subtract in a different variable outside the `if (G4UniformRand() < 0.5)`. The same in `EDPM_SEAL`.